src/Elastic.Markdown/Myst/CodeBlocks/EnhancedCodeBlockParser.cs

```diff
@@ -128,9 +128,38 @@ private static void ProcessAppliesToDirective(AppliesToDirective appliesToDirect
 		}
 	}
 
-	private static void ProcessCallOuts(StringLineGroup lines, string language, EnhancedCodeBlock codeBlock,
+	private static void ProcessCodeBlock(
+		StringLineGroup lines,
+		string language,
+		EnhancedCodeBlock codeBlock,
 		ParserContext context)
 	{
+
+
+		string argsString;
+
+		if (codeBlock.Arguments == null)
+			argsString = "";
+		else if (codeBlock.Info?.IndexOf('{') == -1)
+			argsString = codeBlock.Arguments ?? "";
+		else
+		{
+			var tokens = codeBlock.Arguments.Split();
+			argsString = tokens.Length > 1 ? tokens[1] : "";
+		}
+
+		CodeBlockArguments codeBlockArgs;
+
+		try
+		{
+			codeBlockArgs = new CodeBlockArguments(argsString);
```

Comment:
CodeablockArguments.TryParse with a private constructor on CodeBlockArguments. 

Use the try method pattern instead. 

we have a few good examples of it in the code base. This plays better with null checking too through the not `[NotNullWhen(true)]` annotation.

Reply:
https://github.com/elastic/docs-builder/pull/699/commits/130315d138d49446437b03ea8adc36d354489e68